display_list/display_list_dispatcher.cc

```diff
@@ -137,8 +137,10 @@ static std::optional<Rect> ToRect(const SkRect* rect) {
 
 // |flutter::Dispatcher|
 void DisplayListDispatcher::saveLayer(const SkRect* bounds,
-                                      bool restore_with_paint) {
-  canvas_.SaveLayer(restore_with_paint ? paint_ : Paint{}, ToRect(bounds));
+                                      const flutter::SaveLayerOptions options) {
+  canvas_.SaveLayer(
+    options == flutter::SaveLayerOptions::kWithAttributes ? paint_ : Paint{},
```

Comment:
No, you don't want == here because there may be multiple flags set (do I even implement == on the Options?)

You want options.blah_uses_attributes_blah() which is a method that checks the flag and returns a boolean.

Reply:
If the "can do opacity optimization" flag was also set on this saveLayer then your test would fail.

Reply:
You want `options.renders_with_attributes()`

Reply:
Yeah, you have `==` on `SaveLayerOptions`: https://github.com/flutter/engine/blob/main/display_list/display_list.h#L182

So this should be `options.renders_with_attributes() ? paint_ : Paint{}`?

Reply:
Thanks! Changed to `options.renders_with_attributes()`.